sdk/src/client/secret/ledger_nano.rs

```diff
@@ -296,19 +296,23 @@ impl SecretManage for LedgerSecretManager {
         } else {
             // figure out the remainder address and bip32 index (if there is one)
             let (remainder_address, remainder_bip32): (Option<&Address>, LedgerBIP32Index) =
-                match &prepared_transaction.remainder {
-                    Some(a) => {
-                        let chain = a.chain.ok_or(Error::MissingBip32Chain)?;
-                        (
-                            Some(&a.address),
-                            LedgerBIP32Index {
-                                bip32_change: chain.change.harden().into(),
-                                bip32_index: chain.address_index.harden().into(),
+                prepared_transaction.remainder.as_ref().map_or_else(
```

Comment:
this is less easy to understand :c

Reply:
Clippy complained otherwise, do you want me to put an `#[allow(`?

Reply:
I mean, I guess so. It's really horrible rn.

Reply:
Done